fastestimator/op/numpyop/univariate/rua.py

```diff
@@ -132,274 +158,118 @@ class Posterize(PosterizeAug):
-        super().__init__(inputs=inputs, outputs=outputs, mode=mode, limit=level / 30 * 0.9)
-
-
-@traceable()
-class ShearX(ShearXAug):
-    """Randomly shear the image along the X axis.
-
-    Args:
-        level: Factor to set the range for shear. Must be in the range [0, 30].
-        inputs: Key(s) of images to be modified.
-        outputs: Key(s) into which to write the modified images.
-        mode: What mode(s) to execute this Op in. For example, "train", "eval", "test", or "infer". To execute
-            regardless of mode, pass None. To execute in all modes except for a particular one, you can pass an argument
-            like "!infer" or "!train".
-
-    Image types:
-        uint8
-    """
-    def __init__(self,
-                 level: Union[int, float],
-                 inputs: Union[str, Iterable[str]],
-                 outputs: Union[str, Iterable[str]],
-                 mode: Union[None, str, Iterable[str]] = None):
-        super().__init__(inputs=inputs, outputs=outputs, mode=mode, shear_coef=level / 30 * 0.5)
+                 mode: Union[None, str, Iterable[str]] = None,
+                 num_bits: Union[int,
+                                 Tuple[int, int],
+                                 Tuple[int, int, int],
+                                 Tuple[Tuple[int, int], Tuple[int, int], Tuple[int, int]]] = 7):
+        self.num_bits = num_bits
+        super().__init__(inputs=inputs, outputs=outputs, mode=mode, num_bits=num_bits)
 
+    def set_rua_level(self, magnitude_coef: float) -> None:
+        """Set the augmentation intentity based on the magnitude_coef.
 
-@traceable()
-class ShearY(ShearYAug):
-    """Randomly shear the image along the Y axis.
+        This method is specifically designed to be invoked by RUA Op.
 
-    Args:
-        level: Factor to set the range for shear. Must be in the range [0, 30].
-        inputs: Key(s) of images to be modified.
-        outputs: Key(s) into which to write the modified images.
-        mode: What mode(s) to execute this Op in. For example, "train", "eval", "test", or "infer". To execute
-            regardless of mode, pass None. To execute in all modes except for a particular one, you can pass an argument
-            like "!infer" or "!train".
-
-    Image types:
-        uint8
-    """
-    def __init__(self,
-                 level: Union[int, float],
-                 inputs: Union[str, Iterable[str]],
-                 outputs: Union[str, Iterable[str]],
-                 mode: Union[None, str, Iterable[str]] = None):
-        super().__init__(inputs=inputs, outputs=outputs, mode=mode, shear_coef=level / 30 * 0.5)
+        Args:
+            magnitude_coef: Factor to set the range for magnitude of augmentation.
+        """
+        super().__init__(inputs=self.inputs,
+                         outputs=self.outputs,
+                         mode=self.mode,
+                         num_bits=round(8 - (magnitude_coef * self.num_bits)))
```

Comment:
this multiplication with self.num_bits won't work if self.num_bits is a tuple or tuple of tuples as described in the doc string